spec/requests/legalaid_application_spec.rb

```diff
@@ -8,6 +9,53 @@
       post '/v1/applications'
       expect(response_json['data']['attributes']['application_ref']).not_to be_empty
       expect(response.status).to eql(201)
+      expect(response.content_type).to eql('application/json')
+    end
+  end
+
+  describe 'GET /v1/applications/:id' do
+    let(:applicant_name) { Faker::Name.name }
+    let(:applicant_date_of_birth) { Faker::Date.birthday(18, 100) }
+    let(:applicant) { Applicant.create(name: applicant_name, date_of_birth: applicant_date_of_birth) }
+    let(:legal_aid_application) { LegalAidApplication.create(applicant: applicant) }
+    let(:application_ref) { legal_aid_application.application_ref }
+
+    context 'with a non-existant application reference' do
+      let(:application_ref) { SecureRandom.uuid }
```

Comment:
Before you read below, I'd like to highlight I'm not recommending any change here.  I am using this opportunity to highlight the dangers of flaky tests 👻.

A test which gives false positives, i.e. it randomly fails when there isn't a defect with the production code, causes the devs to lose confidence in the test.  A test which exhibits this behaviour is described as flaky.  Having flaky tests is bad. 

Symptoms of this can include your CI build fails when it was passing locally, re-running the same build on CI causes the build to pass again.

Whenever I see a dynamic expression being generated and used in a test I make sure to confirm that using a dynamic value is:

A) More valuable than using a constant value.
B) Won't create a situation where the test wouldn't randomly fail.

In this situation, the test _could_ fail if the `application_ref` was valid.  An extremely unlikely situation given the nature of UUIDs.

Use of `Faker` above also introduces randomness.  Imagine an absurd situation where a new business rule was introduced which said applicants with the name "Adrian" gave a validation error, this code would become flaky as a result.

Developers should prevent flaky tests from being introduced.  If at a later date flakiness is observed with a test, the fixing of this should be prioritised with high importance.  The longer the flakiness lives in a test suite, the more time will be wasted on re-running tests.

Reading material on flaky tests:
https://testing.googleblog.com/2016/05/flaky-tests-at-google-and-how-we.html
https://blogs.dropbox.com/tech/2018/05/how-were-winning-the-battle-against-flaky-tests/